azerothcore/azerothcore-wotlk
# feat(Core/Warden): optimization + PQR detection

Noting, these changes are started by @Helias  and also inspired by TrinityCore developers work: Treeston. And also thanks to scizzydo for actual ideas (in TrinityCore project)

## CHANGES PROPOSED:
-  Heavy code optimizations and refactoring
-  PQR detection
-  Implemented pending checks queue
-  Improved warden logging
-  Implement new config options to handle LUA checks
-  Removed forced warden checks


## ISSUES ADDRESSED:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/2004



## TESTS PERFORMED:
-  PQR auto rotation
-  Various Wowemuhack style hacks




## HOW TO TEST THE CHANGES:


- Attempt to use PQR based or Wowemuhack based hacks and check if warden detects it (NOTE: some PQR releases with antiwarden cannot be detected, since that completely fakes warden in clientside)

## KNOWN ISSUES AND TODO LIST:



## Target branch(es):








 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
